### ISSUE TITLE
Angle between vectors as an extra function for CustomHBondForce

### ISSUE THREAD

**Issue by cabb99** [openmm/openmm#2566]
Currently the CustomHBondForce can be defined using distance, angles and dihedrals. The forcefield I'm working on requires the implementation of angles between a pair of vector, where each vector is defined by 2 particles, having 4 particles in total. It is possible to calculate the angle between vectors using the following energy expression:
```
t3 = acos(cost3lim);
cost3lim = min(max(cost3,-1),1);
cost3 = sin(t1)*sin(t2)*cos(phi)-cos(t1)*cos(t2);
t1 = angle(d2,d1,a1);
t2 = angle(d1,a1,a2);
phi = dihedral(d2,d1,a1,a2);
```
where the two vectors are d2->d1 and a2->a1.
Doing this introduces a singularity when cost3 is close to -1 or 1. This problem has already been fixed using the computeAngle function in the code. I wonder if it would be possible to implement a new vector angle function.
```t3 = vectorangle(d2,d1,a1,a2);```

**Comment by peastman** [comment]
Yes, that would be possible.  I want to think about this a little and consider what's the most elegant way of handling this.

**Comment by wayuer19** [comment]
Hi, still any possible to implement this kind of vector angle function? Really useful in my case.

**Comment by wayuer19** [comment]
CustomHbondForce. I'm using it to mimic parellal stacking kind force.

**Comment by cabb99** [comment]
#4362

**Comment by cabb99** [comment]
I wrote an implementation of the vector angle function that may be useful. #4362
